### ISSUE TITLE
deviance definition to use for estimating theta in glmer.nb

### ISSUE THREAD

**Issue by aosmith16** [lme4/lme4#287]
I recently switched to using the development version of lme4 (1.1-8) and have run into problems with models fit using `glmer.nb` (which I know is experimental).

I'm using an example from the `glmmadmb` help page to illustrate what I've been experiencing.

```
library(glmmADMB)
# Load development version of lme4
library(lme4, lib.loc = "C:/Program Files/R/Packages")
packageVersion("lme4")
[1] ‘1.1.8’
```
```
# Fit model with most current version of lme4 (1.1-8)
fit1 = glmer.nb(y ~ Base*trt + Age + Visit + (1|subject), data=epil2)
fit1

Generalized linear mixed model fit by maximum
  likelihood (Laplace Approximation) [glmerMod]
 Family: Negative Binomial(0.65)  ( log )
Formula: y ~ Base * trt + Age + Visit + (1 | subject)
   Data: ..2
      AIC       BIC    logLik  deviance  df.resid 
1421.5207 1449.2314 -702.7604 1405.5207       228 
Random effects:
 Groups   Name        Std.Dev. 
 subject  (Intercept) 1.171e-06
 Residual             6.109e-01
Number of obs: 236, groups:  subject, 59
Fixed Effects:
      (Intercept)               Base       trtprogabide  
          -1.2401             0.8790            -0.8726  
              Age              Visit  Base:trtprogabide  
           0.4762            -0.2725             0.3342
```

The estimate of theta is very small and the random effects variance is essentially 0.  Compare this to the output from a model fit with `glmmadmb`.

```
# Compare with glmmadmb fit
epil2$subject = factor(epil2$subject)
fit2 = glmmadmb(y ~ Base*trt + Age + Visit + (1|subject), data=epil2, family="nbinom")
fit2

GLMM's in R powered by AD Model Builder:

  Family: nbinom 
  alpha = 7.4702 
  link = log 

Fixed effects:
  Log-likelihood: -624.551 
  AIC: 1265.102 
  Formula: y ~ Base * trt + Age + Visit + (1 | subject) 
      (Intercept)              Base      trtprogabide 
       -1.3300404         0.8839167        -0.9299655 
              Age             Visit Base:trtprogabide 
        0.4751431        -0.2701602         0.3372419 

Random effects:
Structure: Diagonal matrix
Group=subject
            Variance StdDev
(Intercept)   0.2172  0.466

Number of observations: total=236, subject=59
```

Things look more reasonable to me if I go back to lme4 version 1.1-7.

```
# Detach package lme4 version 1.1-8 and load 1.1-7
detach("package:lme4", unload=TRUE)
library(lme4, lib.loc = "C:/Program Files/R/R-3.1.3/library")
packageVersion("lme4")
[1] ‘1.1.7’

# Refit model with lme4 1.1-7
fit3 = glmer.nb(y ~ Base*trt + Age + Visit + (1|subject), data=epil2)
fit3

Generalized linear mixed model fit by maximum
  likelihood (Laplace Approximation) [glmerMod]
 Family: Negative Binomial(7.4734)  ( log )
Formula: y ~ Base * trt + Age + Visit + (1 | subject)
   Data: ..2
      AIC       BIC    logLik  deviance  df.resid 
1264.9701 1292.6808 -624.4851 1248.9701       228 
Random effects:
 Groups   Name        Std.Dev.
 subject  (Intercept) 0.4505  
 Residual             0.9688  
Number of obs: 236, groups:  subject, 59
Fixed Effects:
      (Intercept)               Base       trtprogabide  
          -1.3380             0.8826            -0.9279  
              Age              Visit  Base:trtprogabide  
           0.4746            -0.2685             0.3365 
```
After poking around for a bit, my best guess is that the `deviance.merMod` function in the development version of lme4 now returns the sum of the squared deviance residuals for GLMM's, where it used to return twice the negative log-likehood.  The `deviance` function is used in the final theta optimization step of `glmer.nb` (in the `optTheta` function).

**Comment by aosmith16** [comment]
After reading up on some of the discussion/clarification on deviance (relative vs absolute, etc.) I see now that the `deviance` function in lme4 version 1.1-8 returns the relative deviance. 

If the optimization of theta should be based on minus twice the log likelihood, maybe `dev <- deviance(lastfit <<- refitNB(lastfit, theta = exp(t), control = control))` in `optTheta` just needs to be replaced by `dev <- -2*logLik(lastfit <<- refitNB(lastfit, theta = exp(t), control = control))`.

**Comment by mmaechler** [comment]
@aosmith16 : Thank you.  You make very convincing points, and I have confirmed all you say.
Unfortunately, we did not have any proper checks for  `glmer.nb` results in 1.1-7 or 1.1-8... and even your proposed remedy, replacing  `deviance(.)` which has *changed* its semantic (for 1.1-8 .. and it seems we did *NOT* systematically check the use of `deviance()` in our own sources... BLUSH!!).. again replacing `deviance(.)`  by `-2 * logLik(.)`  does indeed revert to 1.1-7 behaviour (well, modulo the fact that we've also tweaked the optimization defaults) which looks  *much* better.

I'm going to commit the above fix to `glmer.nb()` in a moment.
